InnerEye/ML/SSL/datamodules_and_datasets/transforms_utils.py

```diff
@@ -0,0 +1,280 @@
+    are called on. If False, simply returned one transformed version of the sample.
+    """
+    train_transforms = create_chest_xray_transform(config, is_train=True)
+    val_transforms = create_chest_xray_transform(config, is_train=False)
+    if return_two_views_per_sample:
+        train_transforms = DualViewTransformWrapper(train_transforms)
+        val_transforms = DualViewTransformWrapper(val_transforms)
+    return train_transforms, val_transforms
+
+
+def create_chest_xray_transform(config: CfgNode,
+                                is_train: bool) -> Callable:
+    """
+    Defines the image transformations pipeline used in Chest-Xray datasets.
+    Type of augmentation and strength are defined in the config.
+    :param config: config yaml file fixing strength and type of augmentation to apply
+    :param is_train: if True return transformation pipeline with augmentations. Else, disable augmentations i.e.
+    only resize and center crop the image.
+    """
+    transforms: List[Any] = []
+    if is_train:
+        if config.augmentation.use_random_affine:
+            transforms.append(RandomAffine(config))
+        if config.augmentation.use_random_crop:
+            transforms.append(RandomResizeCrop(config))
+        else:
+            transforms.append(Resize(config))
+        if config.augmentation.use_random_horizontal_flip:
+            transforms.append(RandomHorizontalFlip(config))
+        if config.augmentation.use_gamma_transform:
+            transforms.append(RandomGamma(config))
+        if config.augmentation.use_random_color:
+            transforms.append(RandomColorJitter(config))
+        if config.augmentation.use_elastic_transform:
+            transforms.append(ElasticTransform(config))
+        transforms += [CenterCrop(config), ToTensor()]
+        if config.augmentation.use_random_erasing:
+            transforms.append(RandomErasing(config))
+        if config.augmentation.add_gaussian_noise:
+            transforms.append(AddGaussianNoise(config))
+    else:
+        transforms += [Resize(config), CenterCrop(config), ToTensor()]
+    transforms.append(ExpandChannels())
+    return torchvision.transforms.Compose(transforms)
+
+
+class InnerEyeCIFARTrainTransform(SimCLRTrainDataTransform):
+    """
+    Overload lightning-bolts SimCLRTrainDataTransform, to avoid return unused eval transform.
+    """
+
+    def __call__(self, sample: Any) -> Tuple[Any, Any]:
+        transform = self.train_transform
+        xi = transform(sample)
+        xj = transform(sample)
+        return xi, xj
+
+
+class InnerEyeCIFARLinearHeadTransform(SimCLRTrainDataTransform):
+    """
+    Overload lightning-bolts SimCLRTrainDataTransform, to avoid return unused eval transform.
+    """
+
+    def __call__(self, sample: Any) -> Any:
+        return self.online_transform(sample)
+
+
+class InnerEyeCIFARValTransform(SimCLREvalDataTransform):
```

Comment:
It looks like this is exactly the same class as the one above (l254-263)

Reply:
indeed, cleaned up